test/image/mocks/multicategory-sorting.json

```diff
@@ -0,0 +1,37 @@
+{
+    "data": [{
+        "x": [[1, 2, 3, 4, 5 ,6],
+              [1, 2, 1, 2, 1 ,2]],
+        "y":  [1, 2, 3, 4, 5 ,6]
+    },
+
+    {
+        "x": [[1, 2, 3, 4, 5, 6],
+              [1, 2, 1, 2, 1, 2]],
+        "y":  [1, 2, 3, 4, 5, 6],
+        "xaxis": "x2",
+        "yaxis": "y2"
+    },
+    {
+        "x": [[4, 5, 6, 7, 8, 9],
+              [1, 2, 1, 2, 1, 2]],
+        "y":  [1, 2, 3, 4, 5, 6],
+        "xaxis": "x2",
+        "yaxis": "y2"
+    },
+
+    {
+        "x": [[1, 2, 3, 4, 5, 6, 4, 5, 6, 7, 8, 9],
+              [1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2]],
+        "y":  [1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6],
+        "xaxis": "x3",
+        "yaxis": "y3"
+    }],
```

Comment:
@etpinard Would you please adding these two traces to the end of this mock? I was curious about the `null` cases. But your algorithm was able to ignore them. 
<pre>
{
        "x": [[1, 2, 3, 4, 5, 6],
              [1, 2, 1, 2, null, 2]],
        "y":  [1, 2, 3, 4, 5, 6],
        "xaxis": "x4",
        "yaxis": "y4"
    },
    {
        "x": [[4, 5, 6, 7, 8, 9],
              [1, null, 1, 2, 1, 2]],
        "y":  [1, 2, 3, 4, 5, 6],
        "xaxis": "x4",
        "yaxis": "y4"
    }
</pre>

Reply:
Good call! Done in -> 96edd7215